libs/designer-ui/src/lib/editor/base/utils/editorToSegment.ts

```diff
@@ -142,3 +163,19 @@ const collapseLiteralSegments = (segments: ValueSegment[]): void => {
     index++;
   }
 };
+
+const removeNewlinesAndSpaces = (inputStr: string): string => {
+  return inputStr.replace(/\s+/g, '').replaceAll(/\n/g, '').replaceAll(/\r/g, '');
+};
+
+export const getTokenFromNodeMap = (segmentSoFar: string, nodeMap: Map<string, ValueSegment>): string | undefined => {
+  const copyOfSegment = segmentSoFar;
+  const processedId = copyOfSegment.replace(/\n/g, '');
```

Comment:
For consistency, let's have a method called `removeNewlines` that removes both `\n` and `\r`, and use that on lines 173 and 175 here. `removeNewlinesAndSpaces` can also call that helper.

Reply:
Good point. I haven't gone back and optimized my changes yet, so I'll go back and make this change (among a few others I have in mind).